doc/src/modules/solvers/solveset.rst

```diff
@@ -1,12 +1,463 @@
+                => `{x| log(f(x)) + log(g(x));x in S}` otherwise
+
+ * product solve: {x|f(x)*g(x)=0; x in S}
+                => `{x|f(x)=0; x in S} U {x|g(x)=0; x in S}` given f(x) and g(x) are bounded
+                => {x|f(x)*g(x)=0; x in S}, otherwise
+
+ Since the output type is same as input type any composition of these
+ transformations is also a valid transformation. And our aim is to find
+ the right sequence of compositions (given the atoms) which transforms
+ the given condition set to a set which is not a condition set i.e.,
+ FiniteSet, Interval, Set of Integers and their Union, Intersection,
+ Complement or ImageSet. We can assign a cost function to each set,
+ such that, more desirable that form of set is to us, the less the value
+ of cost function. This way our problem is now reduced to finding the path
+ from the initial ConditionSet to the lowest valued set on a graph where
+ the atomic transformations forms the edges.
+
+
+How do we deal with cases where only some of the solutions are known?
+---------------------------------------------------------------------
+
+ Creating a Universal equation solver, which can solve each and every
+ equation we encounter in mathematics is an ideal case for solvers in
+ a Computer Algebra System. We always have some cases, which are not
+ solved, or solved with incomplete solutions, so it's very important
+ to represent that situation. For this type of situation we use
+ `ConditionSet` class in the sets module, which acts as an unevaluated
+ solveset object.
+
+ ConditionSet is basically a Set of elements which satisfies a given
+ condition. For example, to represent the solutions of the Equation
+ in Real domain:
+
+ .. math::  (x^2 - 4)*(sin(x) + x)
+
+ We can represent it as:
+
+ `{-2, 2} ∪ {x | x ∊ ℝ ∧ x + sin(x) = 0}`
+
+
+What will you do with the old solve?
+------------------------------------
+
+ There is still a few things `solveset` can't do, which the old `solve`
+ can, such as solving non linear multivariate & LambertW type equations.
+ Hence, it's not yet a perfect replacement for old `solve`. The ultimate
+ goal is to:
+
+ * Replace ``solve`` with ``solveset`, by the time solveset is
+   atleast powerful as ``solve``, i.e. ``solveset`` does everything
+   that ``solve`` can do currently, and
+
+ * Eventually rename ``solveset`` as ``solve``. Meanwhile
+   issue a deprecation warning for the current behavior of ``solve``.
+
+
+How are symbolic parameters handled in solveset?
+------------------------------------------------
+
+ Solveset is in it's initial phase of development as of now, so the
+ symbolic parameters aren't handled well for all the cases, but some
+ work has been done in this regard to depict our ideology towards
+ symbolic parameters. As an instance the solving of `|x| = n` for `x`
+ where `n` is a symbolic parameter.
+
+ Solveset returns the value of `x` considering the domain of the symbolic
```

Comment:
This should be part of the previous paragraph (or the last sentence of the previous paragraph should be with this one).